### ISSUE TITLE
Popup "A monitor scale change was detected" every few minutes

### ISSUE THREAD

**Issue by fixablecar** [spyder-ide/spyder#11846]
## Problem Description

<img width="419" alt="2020-03-18 2 09 05" src="https://user-images.githubusercontent.com/5638823/76931614-090ff000-6924-11ea-9fc2-1fe1c18634c8.png">

After upgrading Spyder to 4.1.0 through conda, Spyder will show a popup stating that a monitor scale change was detected every few minutes. 

In the popup, there is a recommendation "**Enable auto high DPI scaling**" in Spyder preference will be helpful.

However, even if I enable that, and restart my Spyder, the popup still appears every few minutes.

~~**Actually, I have a workaround to this issue, which is **disabling** the _Automatic Graphics Switching_ in  _System Preference_ -> _Energy Saver_ .**~~
I thought it helps, but still the popup keeps appearing.

## Versions

* Spyder version: 4.1.0
* Python version: 3.7.6 64-bit
* Qt version: 5.9.6
* PyQt version: 5.9.2
* Operating System name/version: Darwin 19.3.0

I am using MacBook Pro 2018 with Catalina, and with no external display.

**Comment by SergeiDBykov** [comment]
I have such a problem too. Moreover, the windows appears is I make a click anywhere outside spyder, on the other desktop or another monitor. 
Darwin 16.7.0

**Comment by ccordoba12** [comment]
Are you changing monitors constantly? Or moving Spyder among monitors?

**Comment by fixablecar** [comment]
> Are you changing monitors constantly? Or moving Spyder among monitors?

No, I don’t have external display.
I will try with external monitor to see if issue persist.

**Comment by ccordoba12** [comment]
@dalthviz, I don't understand why this happens but I think we have two options here:

1. See if there's a better way to detect a screen resolution change using Qt.
2. If not, we need to offer the possibility of not showing this message during the entire session.

**Comment by Colin-Gnt** [comment]
Hi, same problem here.
I don't have a second monitor but it happend when I'm in full screen with spyder and I switch space on my mac for working on another app.

**Comment by ccordoba12** [comment]
@fixablecar, @SergeiDBykov, @Colin-Gnt, are all of you using macOS?

**Comment by Colin-Gnt** [comment]
I am.

**Comment by steff456** [comment]
I can reproduce it by having a spyder in fullscreen mode

**Comment by SergeiDBykov** [comment]
> @fixablecar, @SergeiDBykov, @Colin-Gnt, are all of you using macOS?

Yes, MacOS 10.12

**Comment by fixablecar** [comment]
> @fixablecar, @SergeiDBykov, @Colin-Gnt, are all of you using macOS?

yes, as stated, Macbook Pro 2018 with macOS Catalina.

**Comment by andrewssobral** [comment]
Same problem for me. It occurs in fullscreen mode, when we switch the desktops.

<img src=https://user-images.githubusercontent.com/483114/77090949-2e762a00-6a08-11ea-8741-0c04236bb67a.png width="33%" />

I was obliged to downgrade to 4.0.1:
```
The following packages will be DOWNGRADED:

  spyder                                       4.1.0-py37_0 --> 4.0.1-py37_0
  spyder-kernels                               1.9.0-py37_0 --> 1.8.1-py37_0


Proceed ([y]/n)? 

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
```

**Comment by chenge200x** [comment]
I encountered the same problem. Tried everything and it seems for me that the only cure is to downgrade Spyder.

**Comment by stephenchenxj** [comment]
ya, it is very annoying. 

solution: quit fullscreen mode, then no more popups.

**Comment by juanis2112** [comment]
@ccordoba12  let me know when you close the PR so I can check if it got fixed

**Comment by ccordoba12** [comment]
Ok, wiil do.

**Comment by akrassnigg** [comment]
I have the same issue. MacBook Pro 13-inch, 2019 with Catalina.

Tracked it down to every time the screensaver starts. Actually, it is enough to go to the system preferences and open the screensaver settings (where the screensaver is started in a little preview window).

Entering and exiting fullscreen mode triggers it for me as well, but I hadn't used that before.

**Comment by donkasper** [comment]
I have the same issue, MacBook Pro 13-inch, 2019 with Catalina and using extended display to an external monitor. The issue happens for me whenever I click away from spyder to another program and then back to spyder. Auto-scaling is enabled and I also have restarted the program many times but it comes right back. Screen can be full-mode or not and the issue still happens.

**Comment by ccordoba12** [comment]
Thanks for the info @donkasper.

**Comment by HLiuUS** [comment]
> I have the same issue, MacBook Pro 13-inch, 2019 with Catalina and using extended display to an external monitor. The issue happens for me whenever I click away from spyder to another program and then back to spyder. Auto-scaling is enabled and I also have restarted the program many times but it comes right back. Screen can be full-mode or not and the issue still happens.

Exactly the same thing here for me, very annoying.

**Comment by ccordoba12** [comment]
I've locked this issue to not receive more "me too" comments, which are really unhelpful.

We decided to deactivate this warning entirely for macOS because it's being triggered in a lot of unexpected situations. However, on Windows and Linux is working correctly.

**Comment by ccordoba12** [comment]
This error will be fixed in our next version (4.1.2). In the meantime, please downgrade to our 4.0.1 version with these commands:

    conda install spyder=4.0.1
    conda install spyder-kernels=1.8.1
